### DISCUSSION TITLE
Can I show all broken links ?

### DISCUSSION THREAD

**Discussion by lyndhurst** [zk-org/zk-nvim#148] [Help]
### Verify

- [X] I searched the existing discussions for help

### How can we help you?

Hi,
I was wondering if it was possible to show all broken links in a convenient way to do some notebook maintenance.

I tried showing all workspace 'errors' diagnostics. I tried with `trouble` or the commands I could find in the help, but all I can show are the buffer diagnostics. I figure since note repositories can get pretty large, maybe it is by design, not to load the lsp server unnecessarily.

Maybe someone uses some `zk list` magic that I could load in the quickfix window, or even in a finder.

Thanks in advance for your help.

**Comment by tjex** [comment]
there isn't any inbuilt way to do this that I know. 
Yep, lsp will trigger for each buffer only by design.

The only hackery that comes to mind is to use the CLI commands for `zk`, along the lines of:

`zk list --linked-by . | <get all other documents> | grep '\[\['`

This would get all notes with no outgoing links, and grep them for link syntax (as zk wouldn't find any outgoing links if they were broken)

**Reply by tjex** [reply]
I agree though that this would be a good feature
